src/bytebuf.rs

```diff
@@ -131,6 +131,17 @@ impl BorrowMut<Bytes> for ByteBuf {
     }
 }
 
+impl<T> From<T> for ByteBuf
+where
+    T: Into<Vec<u8>>,
```

Comment:
This impl looks too general to me. For example, this would block us from adding `impl From<ByteBuf> for Vec<u8>`.

Reply:
Indeed. I made some adjustments.